conans/client/build/autotools_environment.py

```diff
@@ -188,7 +188,7 @@ def _configure_link_flags(self):
     def _configure_flags(self):
         ret = copy.copy(self._deps_cpp_info.cflags)
         ret.append(self._architecture_flag)
-        if self._build_type == "Debug":
+        if self._build_type == "Debug" and str(self._compiler) in ['gcc', 'clang', 'apple-clang', 'sun-cc']:
```

Comment:
Maybe use ``str(self.compiler) != "Visual Studio"``? 
Just in case someone adds another compiler, like ICC, or arm-gcc to their own settings, I'd say the chances to accept those flags might be higher than the opposite.

BTW, what is exactly the scenario of using AutoToolsBuildEnvironment with Visual Studio? Use configure/make, but underlying ``cl.exe``? But typically ``cl.exe`` won't understand the syntax of gcc, that is the one provided by this helper.

Reply:
I prefer white-list over black-list - add flags only for known compilers.
but if you insist, it's easy to change.

Reply:
Yes, that makes sense too, leave it then.